### PR TITLE
Query block: Expose initial templates as block variations

### LINKED ISSUE #26194: Expose initial templates as variations
When inserting a Query block we should explore offering a few starting variations:

- Title and Date
- Title and Excerpt
- Title, Date, and Excerpt
- Title and Content
- Title, Featured Image, something else?

We might also want to expose some part of the query config as a second step.

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description
Resolves: https://github.com/WordPress/gutenberg/issues/26194

This PR expose initial templates like `Title and Excerpt`, `Title and Featured Image` as block variations of the `Query` block.
For now it uses the `Placeholder` pattern like `Columns` and these variations are not exposed in the inserter.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/block-library/src/query/edit/index.js
Comment (mtias) [inline packages/block-library/src/query/edit/index.js:24]: Wouldn't this make more sense as `QueryContent`?

**Comment by mapk** [comment]
Hey @ntsekouras, thanks for getting this PR underway!

I took a screenshot of what I'm seeing, and sharing it as a visual representation of where this PR is currently. I know the icons are just placeholders right now, so I went ahead and created some of the other SVGs so you can use those instead.

**Screenshot**

Screen Shot 2020-10-22 at 1 44 10 PM


### Title & Date
Screen Shot 2020-10-22 at 1 38 35 PM

**SVG**
```
<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34 30">
  <Path d="M34 0H0v3h34V0zM12 5H0v1h12V5zM0 17h12v1H0v-1zm34-5H0v3h34v-3zM0 29h12v1H0v-1zm34-5H0v3h34v-3z" />
</SVG>
```

### Title & Excerpt
title-excerpt

**SVG**
```
<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34 30">
  <Path d="M34 0H0v3h34V0zm-4 5H0v1h30V5zm4 3H0v1h34V8zM0 11h30v1H0v-1zm0 12h30v1H0v-1zm34 3H0v1h34v-1zM0 29h30v1H0v-1zm34-11H0v3h34v-3z" />
</SVG>
```

### Title, Date & Excerpt
title-date-excerpt

**SVG**
```
<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34 30">
  <Path d="M34 0H0v3h34V0zM12 5H0v1h12V5zm22 3H0v1h34V8zM0 11h34v1H0v-1zm0 12h12v1H0v-1zm34 3H0v1h34v-1zM0 29h34v1H0v-1zm34-11H0v3h34v-3z" />
</SVG>
```

### Title & Content
title-content

**SVG**
```
<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34 30">
  <Path d="M0 5h30v1H0zM0 17h30v1H0zM0 11h30v1H0zM0 23h30v1H0zM0 8h34v1H0zM0 20h34v1H0zM0 14h34v1H0zM0 26h34v1H0zM0 29h34v1H0zM0 0h34v3H0z"/>
</SVG>
```

File: packages/block-library/src/query/edit/index.js
Comment (ntsekouras) [reply packages/block-library/src/query/edit/index.js:24]: Yes it would. I just made the draft PR to look how it feels and will polish it now.

**Comment by ntsekouras** [comment]
Thanks @mapk ! I've added them and looks nice :) 

Do you think we should make one more variation to start with including `Post Featured Image`? Just a simple one with
`Title`, `Featured Image` in a vertical layout (title and below the image). If yes it would be great if you make an icon as well.

**Comment by mtias** [comment]
I really want to include at least one with featured image. We can remove some of the others if necessary.

**Comment by mtias** [comment]
I don't think these icons math the general visual flair of our icons super well, but let's merge this as a start anyways. We can also continue to refine the specific templates offered further.

Another thing that comes to mind is that instead of a series of bundles we could offer the user to customize with checkboxes like:

- [x] Title
- [x] Date
- [ ] Featured Image
- [ ] Excerpt
- [x] Content

(This is just a thought to consider and should not block this PR.)

File: packages/block-library/src/query/edit/index.js
Comment (gziolo) [inline packages/block-library/src/query/edit/index.js:81]: I'm surprised it works out of the box when replicated from Columns block. Let's see if this is something that could be further abstracted as proposed in https://github.com/WordPress/gutenberg/issues/20582.

File: packages/block-library/src/query/icons.js
Comment (gziolo) [inline packages/block-library/src/query/icons.js:30]: This line is redundant because those icons are imported directly from exported constants in another place.

File: packages/e2e-tests/specs/editor/various/autosave.test.js
Comment (gziolo) [inline packages/e2e-tests/specs/editor/various/autosave.test.js:305]: I guess this one shouldn't be part of this PR.

File: packages/block-library/src/query/edit/query-placeholder.js
Comment (gziolo) [inline packages/block-library/src/query/edit/query-placeholder.js:49]: Columns block has the option to skip selection and go with the default setup. Should we add it here as well? See:

https://github.com/WordPress/gutenberg/blob/263c5ace7e7cb4b79c80dae18e58c6e5dc788a71/packages/block-library/src/columns/edit.js#L242

**Review by gziolo — APPROVED** [review]
Great one. I'm glad it works without too many hassles by replicating the Columns block. I left some comments but I don't consider them blockers. 

I tested new behavior and it all works as expected, including proper initialization after page reload.

**Comment by mapk** [comment]
I'd like to have the icons limited to 34px in width (just like the Columns block) while the button itself remains at 48px before this gets merged. The result should look like this.

Screen Shot 2020-10-23 at 7 52 34 AM

@ntsekouras, I remember you mentioning that the grid formation with the featured image doesn't work right now, so I can get you another icon that just shows the featured image stacked on top of the other text.

**Comment by mapk** [comment]
@ntsekouras here is an SVG for the featured image option.

### Title, Date & Featured Image
Screen Shot 2020-10-23 at 8 02 08 AM

**SVG**
```
<SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 34 30">
  <Path d="M0 0h34v6H0V0zm12 8H0v1h12V8zm18 3H0v1h30v-1zm0 18H0v1h30v-1zM0 26h12v1H0v-1zm34-8H0v6h34v-6z" />
</SVG>
```

File: packages/block-library/src/query/edit/query-placeholder.js
Comment (mapk) [reply packages/block-library/src/query/edit/query-placeholder.js:49]: I'm not sure what problem the "Skip" option solves. Forcing a selection of one of the options seems perfectly reasonable to me.

**Comment by mapk** [comment]
>I don't think these icons math the general visual flair of our icons super well, but let's merge this as a start anyways.

@mtias I'm happy to iterate on icons. Currently, their width should match the Columns icons and be limited to 34px. Both Column layout choices and Query layout choices would then still remain wider than the rest of our icons which are set at 24px, mostly.

File: packages/block-library/src/query/edit/index.js
Comment (ntsekouras) [reply packages/block-library/src/query/edit/index.js:81]: Yes, I agree with that and saw that some logic might be extracted, but I think it's better for a follow up.

Another thing I wanted to wait before looking into this, is that for Query the `Placeholder` will be probably be a `steps` init block procedure. At first select some `'layout`(still block variation technically) of the block and then a second step to select some other block options. So I'd like how this could work and be abstracted for other blocks as well.

File: packages/e2e-tests/specs/editor/various/autosave.test.js
Comment (ntsekouras) [reply packages/e2e-tests/specs/editor/various/autosave.test.js:305]: Yes, I just pushed this to check CI on GH. I'll remove it as it's merged here: https://github.com/WordPress/gutenberg/pull/26416

File: packages/block-library/src/query/edit/query-placeholder.js
Comment (ntsekouras) [reply packages/block-library/src/query/edit/query-placeholder.js:49]: I also think that skip is not necessary in Query, for now at least, as it's more hustle to add/remove blocks from Query later, than just selecting a variation that might suit the user more.

**Pull request merged**
